.travis.yml

```diff
@@ -10,35 +10,51 @@ branches:
 language: node_js
 
 node_js:
-  - '10'
-  - '12'
+  - 10
+  - 12
 
 before_install:
-# Add `aws` CLI tool.
+  # Add `aws` CLI tool.
   - pip install --user awscli
 
 install:
   - yarn install --frozen-lockfile --non-interactive
 
 jobs:
   include:
-    - stage: Build Reason
+    - &buildre
+      stage: Build Reason
       script:
         - yarn build:re
-    - stage: Build TypeScript
+    - <<: *buildre
+      node_js: 12
+    - &buildts
+      stage: Build TypeScript
       script:
         - yarn check:ts
-    - stage: Lint
+    - <<: *buildts
+      node_js: 12
+    - &lint
+      stage: Lint
       script:
         - yarn lint
-    - stage: Test
+    - <<: *lint
+      node_js: 12
+    - &testall
+      stage: Test All
       script:
+        - yarn build:re
```

Comment:
Because we have to build reason here for tests, it seems inefficient to also do a `buildre` stage that is essentially thrown away because we don't share artifacts. Maybe change this to `"Build reason and test"` or something?